recipes/hdf5/all/conanfile.py

```diff
@@ -63,12 +66,15 @@ def validate(self):
                 self.info.options.szip_encoding and \
                 not self.dependencies["szip"].options.enable_encoding:
             raise ConanInvalidConfiguration("encoding must be enabled in szip dependency (szip:enable_encoding=True)")
+        if self.settings.get_safe("compiler.cppstd"):
+            check_min_cppstd(self, self._v1_14_minimum_cpp_standard)
 
     def configure(self):
         if self.options.shared:
             self.options.rm_safe("fPIC")
         self.settings.rm_safe("compiler.libcxx")
-        self.settings.rm_safe("compiler.cppstd")
+        if Version(self.version) < "1.14.0" or not self.options.enable_cxx:
+            self.settings.rm_safe("compiler.cppstd")
```

Comment:
why only cppstd and not libcxx, and why is it always removed if version < 1.14? I mean, there is this option enable_cxx in these old versions as well.

Reply:
I agree that this looks a bit odd.

Apparently C++11 was not required when building the C++ wrappers.
Excerpt from the [changelog](https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-1.14/hdf5-1.14.0/src/hdf5-1.14.0-RELEASE.txt) for v1.14.0:

```
- A C++11-compliant compiler is now required to build the C++ wrappers

      CMAKE_CXX_STANDARD is now set to 11 when building with CMake and
      -std=c++11 is added when building with clang/gcc via the Autotools.

      (DER - 2021/05/27)
```

I can try to simplify a bit the logic by never removing `compiler.cppstd` when `self.options.enable_cxx == True`.

I am not sure why `compiler.libcxx` was/is always removed. I think it would be good to keep it when `self.options.enable_cxx == True`

Reply:
Actually it's a regression coming from https://github.com/conan-io/conan-center-index/pull/13605. Yes you can remove libcxx & cppstd only when enable_cxx is False.